81-question-mark-chrysalis/technical_specification.md

```diff
@@ -0,0 +1,230 @@
+**Epic Type:** Implementation Epic
+
+Epic planning and implementation follow the
+[Epic Planning and Marathon SOP](https://ghga.pages.hzdr.de/internal.ghga.de/main/sops/development/epic_planning/).
+
+## Scope
+
+### Outline:
+
+The goal of this epic is to implement a service for the transformation of Experimental Metadata (EM) from one representation/model to another. 
+This service shall provide functionality around the configurable workflow concept from the `metldata` library to enable these transformations.
+To this goal, it needs to keep track of the transformation workflows, original and derived data and workflow routes in its database.
+
+
+![em-transformation-service overview](./images/em_transformation_service.png)
+
+
+### Terminology
+
+`Experimental Metadata (EM)`: Data describing the experimental process of generating the Research Data archived in GHGA.
+
+`Experimental Metadata Ingress Models (EMIM)`: Schemas that define the structure and format of experimental metadata as it enters the GHGA system from external sources. Currently, GHGA has a single data ingress model for experimental metadata, which is the ghga-metadata-schema. 
+
+`Universal Discovery Model`: A common target representation to which all the EMs are transformed into, used for data queries and data display via th GHGA Data Portal.  
+
+`EMPack`: Experimental metadata in datapack format. It follows a relation data schema and compatible with one of the EMIMs.
+
+`AnnotatedEMPack`: A datapack enriched with additional information. 
+
+`AnnotatedEMPacks`: A MongoDB collection of AnnotatedEMPack objects.
+
+`Workflow`: Instructions on how to produce a certain data/model representation from another data/model. The workflows are defined by the `metldata` library on which the transformation service will be built.
+
+`Workflows`: A MongoDB collection of Workflow objects.
+
+`WorkflowRoute`: Information on which specific workflow is used to produce data compliant with the output model when presented with data compliant with the input model.
+
+`WorkflowRoutes`: A MongoDB collection of WorkflowRoute objects.
+
+
+### Included/Required:
+
+- Implement logic to process incoming AnnotatedEMPacks, execute the corresponding workflows and store the resulting AnnotatedEMPacks if required.
+- Implement service start-up logic to derive missing schemas for workflow routes with empty output schemas.
+- Implement MongoDB collections to store AnnotatedEMPacks, Workflows and WorkflowRoutes.
+
+#### Collections
+
+
+`Schemas`
+
+- purpose: store schemas
+
+data structure:
+```python
+class Schema(BaseModel):
+   id: uuid
+   name: str
+   original: bool
+   version: str | None
+   content: SchemaPack | None
+   publish: bool 
+   order: int | None
+
+```
+
+`Workflows`
+
+- Purpose: store metldata-compatible workflow definitions used to transform schemas/data.
+
+data structure:
+```python
+class TransformationWorkflow(BaseModel):
+   id: uuid
+   name: str | None
```

Comment:
I would make the name mandatory as well. It should be a name indicating the purpose of the workflow. Otherwise it will be hard to debug and understand what the operations do.